homeassistant/components/xiaomi_miio/fan.py

```diff
@@ -1103,6 +1161,63 @@ async def async_set_dry_off(self):
         )
 
 
+class XiaomiAirHumidifierMiot(XiaomiAirHumidifier):
+    """Representation of a Xiaomi Air Humidifier (MiOT protocol)."""
+
+    @property
+    def speed(self):
+        """Return the current speed."""
+        if self._state:
+            return AirhumidifierMiotOperationMode(self._state_attrs[ATTR_MODE]).name
```

Comment:
This property may only return one of the official fan speed constants.

Reply:
Is it new rule? All xiaomi models pur/hum around has custom modes
And this is fine, because official is not enought for that. 
https://github.com/rytilahti/python-miio/blob/master/miio/airpurifier_miot.py#L54
https://github.com/rytilahti/python-miio/blob/master/miio/airhumidifier.py#L44
How to add custom "Auto", "Favourite" to the offical, which don't contain in official list, but model has that?
I can to do that but, then I'll cut a functionality of the model, that not the fine even for me. Because I use the hum, how I can to set Auto mode or Favourite for my Purifier (if in future re-write the code https://github.com/rytilahti/python-miio/blob/master/miio/airpurifier.py#L20)

Reply:
The current xiaomi_miio integration isn't following our architecture design. We won't allow it to extend this inconsistency with more changes.

If it's not possible to map between the standard speed values and the device speed values, not supporting the fan speed setting and use a custom service instead.

Reply:
Could you provide some example pls. How to do that.

Reply:
Got it. Ok

Reply:
Will be fine if I do and custom service and official speeds?
So official will work as expected, and custom with the extended mode in my case "auto"?

Reply:
How will you map `Auto`, `Silent`, `Favorite` and `Fan` to `off`, `low`, `medium` and `high`?

Reply:
The hum has https://github.com/rytilahti/python-miio/blob/master/miio/airhumidifier_miot.py#L44
"Auto, low, mid, high" modes, they mapped is good except "auto" in my case, and auto could be available in additional service like "set_mode" where will be full list of modes. If I correct understand "official" is need for integration between services like pass the fan to HomeKit like example. So the will works good and in integrations and with a custom input for dashboard via custom service.

Reply:
I ref a diff models before, because I just didn't understand why the model with super custom was passed review, but my PR is stops. But as you explain already, because inconsistency. And I would like to do maximum integrate my device to HA as possible :)

Reply:
Ok. It sounds good to map `low`, `mid` and `high` in the standard set speed service and leave auto for the custom service. :+1: